### PR TITLE
Only display presenter video within presentation area

### PR DESCRIPTION
**Pull request by emaela**
Updated video frontend to only display the presenter's video when user is within a presentation area. To do so, added presenter as a field in townService's PresentationArea and update relevant code to keep the presenter up-to-date. Currently adding tests for the changes.

Summary of changes:
- Add presenterID field to PresentationArea type and class so that the presenter is synced for all townspeople
- Override remove(player) in PresentationArea class so that if the presenter is being removed, the presentation will be reset
- Update SelectDocumentModel so that the presenter (creator of the presentation) is included in the createPresentationArea request body 
- Update PresentationAreaController to include the presenter field in its constructor, and to set the presenter in fromPresentationAreaModel
- Update interactableUpdate in TownController so that the presenter is updated when a PresentationArea is updated
- Add method and hook in TownController for finding a presenter nearby the user (if one exists)
- Update video ParticipantList so that within a presentation area, only the presenter's video is shown

How it looks if you are presenting:

user-presenter

How it looks if you are an audience:
user-audience

### CODE REVIEW

File: frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx [hunk 1]
```diff
@@ -1,56 +1,56 @@
-import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';
```
Comment (IanWoodard) [inline frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx:1]: Why did we switch from single to double quotes?

File: frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx [hunk 2]
```diff
@@ -1,56 +1,56 @@
...
+import Participant from "../Participant/Participant";
+import useSelectedParticipant from "../VideoProvider/useSelectedParticipant/useSelectedParticipant";
 
 const useStyles = makeStyles((theme: Theme) =>
   createStyles({
     container: {
-      overflowY: 'auto',
-      background: 'rgb(79, 83, 85)',
-      gridArea: '1 / 2 / 1 / 3',
+      overflowY: "auto",
```
Comment (IanWoodard) [inline frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx:17]: can we switch back to single quotes?

File: frontend/src/classes/TownController.ts [hunk 3]
```diff
@@ -699,6 +702,17 @@ export default class TownController extends (EventEmitter as new () => TypedEmit
...
+
+  /**
+   * Determine if any presenters are nearby, if player is located at a presentation area
+   */
+  public nearbyPresenter(): PlayerController | undefined {
+    const ourPlayerInteractableID = this.ourPlayer.location.interactableID;
+    const presentationArea = this.presentationAreas.find(
+      (area: PresentationAreaController) => area.id === ourPlayerInteractableID,
+    );
+    return presentationArea ? presentationArea.presenter : undefined;
```
Comment (IanWoodard) [inline frontend/src/classes/TownController.ts:714]:
```suggestion
    return presentationArea?.presenter;
```

File: frontend/src/classes/PresentationAreaController.ts [hunk 4]
```diff
@@ -177,6 +185,7 @@ export default class PresentationAreaController extends (EventEmitter as new ()
       slide: this.slide,
       numSlides: this.numSlides,
       title: this.title,
+      presenterID: this.presenter ? this.presenter.id : undefined,
```
Comment (IanWoodard) [inline frontend/src/classes/PresentationAreaController.ts:188]:
```suggestion
      presenterID: this.presenter?.id,
```

File: townService/src/town/PresentationArea.ts [hunk 5]
```diff
@@ -82,7 +107,7 @@ export default class PresentationArea extends InteractableArea {
   ): PresentationArea {
     const { name, width, height } = mapObject;
     if (!width || !height) {
-      throw new Error(`Malformed viewing area ${name}`);
+      throw new Error(`Malformed presentation area ${name}`);
```
Comment (IanWoodard) [inline townService/src/town/PresentationArea.ts:110]: good catch!

**Review by IanWoodard — APPROVED** [review]
Everything looks good but the video is a bit off-center, Kevin is looking into it so wait on him to merge. Besides that, LGTM! Left a few suggestions. Good work!

File: frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx (hunk 2 above)
Comment (emaela) [reply frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx:17]: Will do

File: frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx (hunk 1 above)
Comment (emaela) [reply frontend/src/components/VideoCall/VideoFrontend/components/ParticipantList/ParticipantList.tsx:1]: I must've done some command for reformatting the file at some point, should be fixed now

**Pull request merged**